app/controllers/cupons_controller.rb

```diff
@@ -12,20 +12,59 @@ def show
     porcentaje
   end
 
+  def new
+    @cupon = Cupon.new
+  end
+
+  def create
+    @cupon = current_user.cupons.new(cupon_params)
+    
+    if @cupon.save
+      redirect_to root_path
+    else
+      render :new, status: :unprocessable_entity
+    end
+  end
+
+  def edit
+    @cupon = Cupon.find(params[:id])
+  end
+
+  def destroy
+    @cupon = Cupon.find(params[:id])
+    @cupon.destroy
+  end
+
   def update
     if request.patch?
       if params[:type] == "increment"
-        if !voted?(1)
+        unless voted?
           @cupon.increment!(:punctuation)
-          set_voted_cookie(1)
+          create_or_update_vote(true)
+        else
+          vote = Vote.find_by(user_id: current_user.id, cupon_id: @cupon.id)
+          if vote.voted == false
+            @cupon.increment!(:punctuation, 1)
+            vote.update(voted: true)
+          end
         end
         respond_to do |format|
           format.json { render json: { punctuation: @cupon.punctuation } }
         end
       elsif params[:type] == "decrement"
-        if !voted?(-1)
+        unless voted?
           @cupon.decrement!(:punctuation)
-          set_voted_cookie(-1)
+          if @cupon.punctuation <= -10
+            @cupon.destroy
+            redirect_to root_path, alert: "El cupón " + @cupon.title + " ha sido eliminado porque su puntuación es igual o menor que -10"
+            return
+          end
```

Comment:
i think this approach is better:
```
if @cupon.punctuation <= -10
  @cupon.destroy
  redirect_to root_path, alert: "El cupón #{@cupon[:title]} ha sido eliminado porque su puntuación es igual o menor que -10"
end
```

Reply:
the return is unnecesssary

Reply:
what do you think?

Reply:
It's necessary so that the method doesn't continue executing after deleting the coupon.

Instead, can I use this code?
```
          if @cupon.punctuation <= -10
            @cupon.destroy
            redirect_to root_path, alert: "El cupón " + @cupon.title + " ha sido eliminado porque su puntuación es igual o menor que -10"
            continue_execution = false
```